### ISSUE TITLE
The logos example doesn't display the entire line on parser errors

### ISSUE THREAD

**Issue by dlight** [zesterer/chumsky#460]
In the [`logos.rs` example](https://github.com/zesterer/chumsky/blob/7467d76e7b5b66864d9167120a1312b208f982c8/examples/logos.rs), whenever there is a *lexer* error (an error coming straight from logos), it shows the whole line as context in ariadne error messages, all the while highlighting the exact span of the error. That is, after making an edit that causes the lexer to fail, it says:

![image](https://github.com/zesterer/chumsky/assets/179028/ee813a16-da84-4c7e-9c19-c068f0ec0307)

But if the error comes from the Chumsky parser itself, it doesn't show any context besides the exact span of the erroring token:

![image](https://github.com/zesterer/chumsky/assets/179028/6772e2d5-6b00-4aa9-b49c-73ed39770df6)

However, in the [`nano_rust.rs` example](https://github.com/zesterer/chumsky/blob/7467d76e7b5b66864d9167120a1312b208f982c8/examples/nano_rust.rs), all errors are shown properly, with the whole line as context (regardless of whether the error happens in the lexer or the parser, and both are written in Chumsky).

What gives?

Also, this might be more appropriate to ask on the [ariadne](https://github.com/zesterer/ariadne) repository, but, how can this be fixed in the `logos.rs` example? I'm trying to understand what's going on `nano_rust.rs` that makes it different, but I don't have a clue. (I mean, I think I'm supposed to make a span for the line, but where is `nano_rust.rs` doing this exactly? And where does `logos.rs` do it, but only for the lexer?)

**Comment by zesterer** [comment]
What error did you introduce that caused the parser to fail but not the lexer? The only thing I can think of would be adding or removing a parentheses, which is likely to confuse the parser due to the extremely relaxed nature of s-expressions.

**Comment by dlight** [comment]
Yes, I removed a parenthesis! Is this important for displaying the entire line?

Since it pointed to a parser error at a `)` token (and reported the span correctly), I expected it to also display the whole line where this token is located.

**Comment by dlight** [comment]
Actually, nevermind, the entire line is composed by a single token `)` (and some whitespace before it). When I add something before the `)` it displays the other thing.

There's no issue here sorry!

**Comment by zesterer** [comment]
No worries. Incidentally, this is a good reason in favour of languages with lots of different kinds of delimiter: it makes it easier for the parser to figure out where inconsistencies originate.
